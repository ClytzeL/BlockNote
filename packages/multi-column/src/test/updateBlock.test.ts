import { describe, expect, it } from "vitest";
import { setupTestEnv, testEditorSchema } from "./setupTestEnv";
import {
  BlockIdentifier,
  BlockNoteEditor,
  PartialBlock,
} from "@blocknote/core";

const getEditor = setupTestEnv();

function updateBlock(
  editor: BlockNoteEditor<
    typeof testEditorSchema.blockSchema,
    typeof testEditorSchema.inlineContentSchema,
    typeof testEditorSchema.styleSchema
  >,
  blockToUpdate: BlockIdentifier,
  update: PartialBlock<
    typeof testEditorSchema.blockSchema,
    typeof testEditorSchema.inlineContentSchema,
    typeof testEditorSchema.styleSchema
  >
) {
  return editor.updateBlock(blockToUpdate, update);
}

describe("Test updateBlock", () => {
  it("Update column list new children", () => {
    updateBlock(getEditor(), "column-list-0", {
      type: "columnList",
      children: [
        {
          type: "column",
          children: [
            {
              type: "paragraph",
              content: "Inserted Column Paragraph",
            },
          ],
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update column list new empty children", () => {
    updateBlock(getEditor(), "column-list-0", {
      type: "columnList",
      children: [
        {
          type: "column",
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update column new children", () => {
    updateBlock(getEditor(), "column-0", {
      type: "column",
      children: [
        {
          type: "paragraph",
          content: "Inserted Column Paragraph",
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update paragraph to column list", () => {
    updateBlock(getEditor(), "paragraph-0", {
      type: "columnList",
      children: [
        {
          type: "column",
          children: [
            {
              type: "paragraph",
              content: "Inserted Column Paragraph",
            },
          ],
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update nested paragraph to column list", () => {
    updateBlock(getEditor(), "nested-paragraph-0", {
      type: "columnList",
      children: [
        {
          type: "column",
          children: [
            {
              type: "paragraph",
              content: "Inserted Column Paragraph",
            },
          ],
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update column to column list", () => {
    updateBlock(getEditor(), "column-0", {
      type: "columnList",
      children: [
        {
          type: "column",
          children: [
            {
              type: "paragraph",
              content: "Inserted Column Paragraph",
            },
          ],
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update paragraph to column", () => {
    updateBlock(getEditor(), "paragraph-0", {
      type: "column",
      children: [
        {
          type: "paragraph",
          content: "Inserted Column Paragraph",
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update nested paragraph to column", () => {
    updateBlock(getEditor(), "nested-paragraph-0", {
      type: "column",
      children: [
        {
          type: "paragraph",
          content: "Inserted Column Paragraph",
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update column list to column", () => {
    updateBlock(getEditor(), "column-list-0", {
      type: "column",
      children: [
        {
          type: "paragraph",
          content: "Inserted Column Paragraph",
        },
      ],
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update column list to paragraph", () => {
    updateBlock(getEditor(), "column-list-0", {
      type: "paragraph",
      content: "Inserted Column Paragraph",
    });

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Update column to paragraph", () => {
    updateBlock(getEditor(), "column-0", {
      type: "paragraph",
      content: "Inserted Column Paragraph",
    });

    expect(getEditor().document).toMatchSnapshot();
  });
});
