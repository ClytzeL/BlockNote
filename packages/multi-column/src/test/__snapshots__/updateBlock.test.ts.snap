// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test updateBlock > Update column list new children 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Inserted Column Paragraph",
                "type": "text",
              },
            ],
            "id": "1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [],
            "id": "3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "2",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test updateBlock > Update column list new empty children 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [],
        "content": undefined,
        "id": "0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [],
            "id": "2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test updateBlock > Update column new children 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Inserted Column Paragraph",
                "type": "text",
              },
            ],
            "id": "0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;
